src/surrogate_heat_driver.cpp

```diff
@@ -659,6 +661,8 @@ double SurrogateHeatDriver::fluid_temperature(std::size_t pin, std::size_t axial
 
 void SurrogateHeatDriver::write_step(int timestep, int iteration)
 {
+  // TODO: Timers deadlock here...
```

Comment:
Were you planning on taking care of this as part of this PR?

Reply:
I spent some time trying to track down the deadlock, but I didn't have any luck.  Is it okay if I raise an Issue for this and defer it to another PR?

Reply:
Sounds good to me